inprocess_cache.go

```diff
@@ -60,8 +62,16 @@ func (l *memLibCache) BatchSet(keyvalues ...interface{}) error {
 	}
 
 	for i := 0; i < len(keyvalues); i += 2 {
-		key := keyvalues[i].(string)
-		value := keyvalues[i+1].([]byte)
+		key, ok := keyvalues[i].(string)
```

Comment:
this code would panic on invalid keys and values, a library creators we should attempt to prevent panics from bubbling up the provided API